src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -1129,118 +1131,143 @@ bool CmdSketcherViewSection::isActive()
+            "fifth of the grid spacing to a grid line to snap."));
+    snapToGrid->setStatusTip(snapToGrid->toolTip());
+}
 
-        gridSizeBox = new Gui::QuantitySpinBox();
-        gridSizeBox->setProperty("unit", QVariant(QStringLiteral("mm")));
-        gridSizeBox->setObjectName(QStringLiteral("gridSize"));
-        gridSizeBox->setMaximum(99999999.0);
-        gridSizeBox->setMinimum(0.001);
+QWidget* GridSpaceAction::createWidget(QWidget* parent)
+{
+    gridShow = new QCheckBox();
 
-        QWidget* gridSizeW = new QWidget(parent);
-        auto* layout = new QGridLayout(gridSizeW);
-        layout->addWidget(gridAutoSpacing, 0, 0, 1, 2);
-        layout->addWidget(sizeLabel, 1, 0);
-        layout->addWidget(gridSizeBox, 1, 1);
+    gridAutoSpacing = new QCheckBox();
 
-        languageChange();
+    snapToGrid = new QCheckBox();
 
-#if QT_VERSION >= QT_VERSION_CHECK(6,7,0)
-        QObject::connect(gridAutoSpacing, &QCheckBox::checkStateChanged, [this](int state) {
-#else
-        QObject::connect(gridAutoSpacing, &QCheckBox::stateChanged, [this](int state) {
-#endif
-            auto* sketchView = getView();
+    sizeLabel = new QLabel();
 
-            if (sketchView) {
-                auto enable = (state == Qt::Checked);
-                sketchView->GridAuto.setValue(enable);
-            }
-        });
+    gridSizeBox = new Gui::QuantitySpinBox();
+    gridSizeBox->setProperty("unit", QVariant(QStringLiteral("mm")));
+    gridSizeBox->setObjectName(QStringLiteral("gridSize"));
+    gridSizeBox->setMaximum(99999999.0);
+    gridSizeBox->setMinimum(0.001);
 
-        QObject::connect(gridSizeBox,
-                         qOverload<double>(&Gui::QuantitySpinBox::valueChanged),
-                         [this](double val) {
-                             auto* sketchView = getView();
-                             if (sketchView) {
-                                 sketchView->GridSize.setValue(val);
-                             }
-                         });
+    QWidget* gridSizeW = new QWidget(parent);
+    auto* layout = new QGridLayout(gridSizeW);
+    layout->addWidget(gridShow, 0, 0, 1, 2);
+    layout->addWidget(gridAutoSpacing, 1, 0, 1, 2);
+    layout->addWidget(snapToGrid, 2, 0, 1, 2);
+    layout->addWidget(sizeLabel, 3, 0);
+    layout->addWidget(gridSizeBox, 3, 1);
 
-        return gridSizeW;
-    }
+    languageChange();
 
-private:
-    ViewProviderSketch* getView()
-    {
-        Gui::Document* doc = Gui::Application::Instance->activeDocument();
+    QObject::connect(gridShow, &QCheckBox::stateChanged, [this](int state) {
+        auto* sketchView = getView();
 
-        if (doc) {
-            return dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+        if (sketchView) {
+            auto enable = (state == Qt::Checked);
+            sketchView->ShowGrid.setValue(enable);
         }
+    });
+
+    QObject::connect(gridAutoSpacing, &QCheckBox::stateChanged, [this](int state) {

```

Comment:
since Qt 6.7 you should use `checkStageChanged` and we support multiple versions of Qt so you need to wrap it in conditional, just like it was before.